#include <functional> missing?

Building `bionic` branch on Arch GNU/Linux

Adding `#include <functional>` to the affected files solved the issue for me

I wonder if this was introduced by gcc updates or am i just stupid?
<details>
  <summary>Huge error logs:</summary>

```
In file included from /tmp/unity-scopes-api/include/unity/scopes/internal/QueryCtrlObjectBase.h:21,
                 from /tmp/unity-scopes-api/include/unity/scopes/internal/smartscopes/SSQueryCtrlObject.h:21,
                 from /tmp/unity-scopes-api/src/scopes/internal/smartscopes/SSQueryCtrlObject.cpp:19:
/tmp/unity-scopes-api/include/unity/scopes/internal/AbstractObject.h:46:39: error: ‘std::function’ has not been declared
     void set_disconnect_function(std::function<void()> func) noexcept;
                                       ^~~~~~~~
/tmp/unity-scopes-api/include/unity/scopes/internal/AbstractObject.h:46:47: error: expected ‘,’ or ‘...’ before ‘<’ token
     void set_disconnect_function(std::function<void()> func) noexcept;
                                               ^
/tmp/unity-scopes-api/include/unity/scopes/internal/AbstractObject.h:54:10: error: ‘function’ in namespace ‘std’ does not name a template type
     std::function<void()> disconnect_func_;
          ^~~~~~~~
/tmp/unity-scopes-api/include/unity/scopes/internal/AbstractObject.h:54:5: note: ‘std::function’ is defined in header ‘<functional>’; did you forget to ‘#include <functional>’?
/tmp/unity-scopes-api/include/unity/scopes/internal/AbstractObject.h:25:1:
+#include <functional>
 
/tmp/unity-scopes-api/include/unity/scopes/internal/AbstractObject.h:54:5:
     std::function<void()> disconnect_func_;
     ^~~
make[2]: *** [CMakeFiles/unity-scopes-obj.dir/build.make:244: CMakeFiles/unity-scopes-obj.dir/src/scopes/internal/smartscopes/SSQueryCtrlObject.cpp.o] Error 1
```

```
In file included from /tmp/unity-scopes-api/include/unity/scopes/internal/zmq_middleware/ConnectionPool.h:21,
                 from /tmp/unity-scopes-api/src/scopes/internal/zmq_middleware/ConnectionPool.cpp:19:
/tmp/unity-scopes-api/include/unity/scopes/internal/Reaper.h:39:14: error: ‘function’ in namespace ‘std’ does not name a template type
 typedef std::function<void()> ReaperCallback;
              ^~~~~~~~
/tmp/unity-scopes-api/include/unity/scopes/internal/Reaper.h:39:9: note: ‘std::function’ is defined in header ‘<functional>’; did you forget to ‘#include <functional>’?
/tmp/unity-scopes-api/include/unity/scopes/internal/Reaper.h:29:1:
+#include <functional>
 
/tmp/unity-scopes-api/include/unity/scopes/internal/Reaper.h:39:9:
 typedef std::function<void()> ReaperCallback;
         ^~~
/tmp/unity-scopes-api/include/unity/scopes/internal/Reaper.h:48:24: error: expected ‘)’ before ‘cb’
     Item(ReaperCallback cb) :
         ~              ^~~
                        )
/tmp/unity-scopes-api/include/unity/scopes/internal/Reaper.h:57:5: error: ‘ReaperCallback’ does not name a type
     ReaperCallback cb;                               // Called if timeout expires (application-supplied callback)
     ^~~~~~~~~~~~~~
/tmp/unity-scopes-api/include/unity/scopes/internal/Reaper.h:46:8: warning: ‘struct unity::scopes::internal::reaper_private::Item’ only defines private constructors and has no friends [-Wctor-dtor-privacy]
 struct Item final
        ^~~~
/tmp/unity-scopes-api/include/unity/scopes/internal/Reaper.h:54:5: note: ‘unity::scopes::internal::reaper_private::Item::Item(const unity::scopes::internal::reaper_private::Item&)’ is public, but requires an existing ‘struct unity::scopes::internal::reaper_private::Item’ object
     Item(Item const&) = default;
     ^~~~
/tmp/unity-scopes-api/include/unity/scopes/internal/Reaper.h:174:24: error: ‘ReaperCallback’ has not been declared
     ReapItem::SPtr add(ReaperCallback const& cb);
                        ^~~~~~~~~~~~~~
/tmp/unity-scopes-api/src/scopes/internal/zmq_middleware/ConnectionPool.cpp: In member function ‘std::shared_ptr<zmqpp::socket> unity::scopes::internal::zmq_middleware::ConnectionPool::find(const string&)’:
/tmp/unity-scopes-api/src/scopes/internal/zmq_middleware/ConnectionPool.cpp:75:65: error: no matching function for call to ‘unity::scopes::internal::Reaper::add(unity::scopes::internal::zmq_middleware::ConnectionPool::find(const string&)::<lambda()>)’
     auto ri = reaper_->add([this, endpoint]{ remove(endpoint); });
                                                                 ^
In file included from /tmp/unity-scopes-api/include/unity/scopes/internal/zmq_middleware/ConnectionPool.h:21,
                 from /tmp/unity-scopes-api/src/scopes/internal/zmq_middleware/ConnectionPool.cpp:19:
/tmp/unity-scopes-api/include/unity/scopes/internal/Reaper.h:174:20: note: candidate: ‘unity::scopes::internal::ReapItem::SPtr unity::scopes::internal::Reaper::add(const int&)’
     ReapItem::SPtr add(ReaperCallback const& cb);
                    ^~~
/tmp/unity-scopes-api/include/unity/scopes/internal/Reaper.h:174:20: note:   no known conversion for argument 1 from ‘unity::scopes::internal::zmq_middleware::ConnectionPool::find(const string&)::<lambda()>’ to ‘const int&’
/tmp/unity-scopes-api/src/scopes/internal/zmq_middleware/ConnectionPool.cpp: In member function ‘void unity::scopes::internal::zmq_middleware::ConnectionPool::register_socket(const string&, const std::shared_ptr<zmqpp::socket>&, bool)’:
/tmp/unity-scopes-api/src/scopes/internal/zmq_middleware/ConnectionPool.cpp:124:80: error: no matching function for call to ‘unity::scopes::internal::Reaper::add(unity::scopes::internal::zmq_middleware::ConnectionPool::register_socket(const string&, const std::shared_ptr<zmqpp::socket>&, bool)::<lambda()>)’
     auto ri = idle_timeout ? reaper_->add([this, endpoint]{ remove(endpoint); }) : nullptr;
                                                                                ^
In file included from /tmp/unity-scopes-api/include/unity/scopes/internal/zmq_middleware/ConnectionPool.h:21,
                 from /tmp/unity-scopes-api/src/scopes/internal/zmq_middleware/ConnectionPool.cpp:19:
/tmp/unity-scopes-api/include/unity/scopes/internal/Reaper.h:174:20: note: candidate: ‘unity::scopes::internal::ReapItem::SPtr unity::scopes::internal::Reaper::add(const int&)’
     ReapItem::SPtr add(ReaperCallback const& cb);
                    ^~~
/tmp/unity-scopes-api/include/unity/scopes/internal/Reaper.h:174:20: note:   no known conversion for argument 1 from ‘unity::scopes::internal::zmq_middleware::ConnectionPool::register_socket(const string&, const std::shared_ptr<zmqpp::socket>&, bool)::<lambda()>’ to ‘const int&’
make[2]: *** [CMakeFiles/unity-scopes-obj.dir/build.make:296: CMakeFiles/unity-scopes-obj.dir/src/scopes/internal/zmq_middleware/ConnectionPool.cpp.o] Error 1
```

```
In file included from /tmp/unity-scopes-api/src/scopes/internal/zmq_middleware/ServantBase.cpp:19:
/tmp/unity-scopes-api/include/unity/scopes/internal/zmq_middleware/ServantBase.h:55:18: error: ‘function’ in namespace ‘std’ does not name a template type
     typedef std::function<void(Current const&,
                  ^~~~~~~~
/tmp/unity-scopes-api/include/unity/scopes/internal/zmq_middleware/ServantBase.h:55:13: note: ‘std::function’ is defined in header ‘<functional>’; did you forget to ‘#include <functional>’?
/tmp/unity-scopes-api/include/unity/scopes/internal/zmq_middleware/ServantBase.h:29:1:
+#include <functional>
 
/tmp/unity-scopes-api/include/unity/scopes/internal/zmq_middleware/ServantBase.h:55:13:
     typedef std::function<void(Current const&,
             ^~~
/tmp/unity-scopes-api/include/unity/scopes/internal/zmq_middleware/ServantBase.h:59:45: error: ‘DispatchFunc’ was not declared in this scope
     typedef std::unordered_map<std::string, DispatchFunc> DispatchTable;
                                             ^~~~~~~~~~~~
/tmp/unity-scopes-api/include/unity/scopes/internal/zmq_middleware/ServantBase.h:59:57: error: template argument 2 is invalid
     typedef std::unordered_map<std::string, DispatchFunc> DispatchTable;
                                                         ^
/tmp/unity-scopes-api/include/unity/scopes/internal/zmq_middleware/ServantBase.h:59:57: error: template argument 5 is invalid
/tmp/unity-scopes-api/src/scopes/internal/zmq_middleware/ServantBase.cpp: In member function ‘void unity::scopes::internal::zmq_middleware::ServantBase::dispatch_(const unity::scopes::internal::zmq_middleware::Current&, capnp::AnyPointer::Reader&, unity::scopes::internal::zmq_middleware::capnproto::Response::Builder&)’:
/tmp/unity-scopes-api/src/scopes/internal/zmq_middleware/ServantBase.cpp:58:31: error: request for member ‘find’ in ‘((unity::scopes::internal::zmq_middleware::ServantBase*)this)->unity::scopes::internal::zmq_middleware::ServantBase::dispatch_table_’, which is of non-class type ‘unity::scopes::internal::zmq_middleware::ServantBase::DispatchTable’ {aka ‘int’}
     auto it = dispatch_table_.find(current.op_name);
                               ^~~~
/tmp/unity-scopes-api/src/scopes/internal/zmq_middleware/ServantBase.cpp:59:31: error: request for member ‘end’ in ‘((unity::scopes::internal::zmq_middleware::ServantBase*)this)->unity::scopes::internal::zmq_middleware::ServantBase::dispatch_table_’, which is of non-class type ‘unity::scopes::internal::zmq_middleware::ServantBase::DispatchTable’ {aka ‘int’}
     if (it == dispatch_table_.end())
                               ^~~
make[2]: *** [CMakeFiles/unity-scopes-obj.dir/build.make:426: CMakeFiles/unity-scopes-obj.dir/src/scopes/internal/zmq_middleware/ServantBase.cpp.o] Error 1
```

```
In file included from /tmp/unity-scopes-api/include/unity/scopes/internal/OnlineAccountClientImpl.h:21,
                 from /tmp/unity-scopes-api/src/scopes/internal/OnlineAccountClientImpl.cpp:19:
/tmp/unity-scopes-api/include/unity/scopes/OnlineAccountClient.h:121:18: error: ‘function’ in namespace ‘std’ does not name a template type
     typedef std::function<void(ServiceStatus const&)> ServiceUpdateCallback;
                  ^~~~~~~~
/tmp/unity-scopes-api/include/unity/scopes/OnlineAccountClient.h:121:13: note: ‘std::function’ is defined in header ‘<functional>’; did you forget to ‘#include <functional>’?
/tmp/unity-scopes-api/include/unity/scopes/OnlineAccountClient.h:24:1:
+#include <functional>
 #include <unity/util/NonCopyable.h>
/tmp/unity-scopes-api/include/unity/scopes/OnlineAccountClient.h:121:13:
     typedef std::function<void(ServiceStatus const&)> ServiceUpdateCallback;
             ^~~
/tmp/unity-scopes-api/include/unity/scopes/OnlineAccountClient.h:127:38: error: ‘ServiceUpdateCallback’ has not been declared
     void set_service_update_callback(ServiceUpdateCallback callback);
                                      ^~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/unity-scopes-api/src/scopes/internal/OnlineAccountClientImpl.cpp:19:
/tmp/unity-scopes-api/include/unity/scopes/internal/OnlineAccountClientImpl.h:66:59: error: ‘unity::scopes::OnlineAccountClient::ServiceUpdateCallback’ has not been declared
     void set_service_update_callback(OnlineAccountClient::ServiceUpdateCallback callback);
                                                           ^~~~~~~~~~~~~~~~~~~~~
/tmp/unity-scopes-api/include/unity/scopes/internal/OnlineAccountClientImpl.h:108:26: error: ‘ServiceUpdateCallback’ in ‘class unity::scopes::OnlineAccountClient’ does not name a type
     OnlineAccountClient::ServiceUpdateCallback callback_;
                          ^~~~~~~~~~~~~~~~~~~~~
/tmp/unity-scopes-api/src/scopes/internal/OnlineAccountClientImpl.cpp:381:80: error: variable or field ‘set_service_update_callback’ declared void
 void OnlineAccountClientImpl::set_service_update_callback(OnlineAccountClient::ServiceUpdateCallback callback)
                                                                                ^~~~~~~~~~~~~~~~~~~~~
/tmp/unity-scopes-api/src/scopes/internal/OnlineAccountClientImpl.cpp:381:80: error: ‘ServiceUpdateCallback’ is not a member of ‘unity::scopes::OnlineAccountClient’
/tmp/unity-scopes-api/src/scopes/internal/OnlineAccountClientImpl.cpp: In lambda function:
/tmp/unity-scopes-api/src/scopes/internal/OnlineAccountClientImpl.cpp:627:13: error: ‘callback_’ was not declared in this scope
         if (callback_)
             ^~~~~~~~~
/tmp/unity-scopes-api/src/scopes/internal/OnlineAccountClientImpl.cpp:627:13: note: suggested alternative: ‘callback’
         if (callback_)
             ^~~~~~~~~
             callback
make[2]: *** [CMakeFiles/unity-scopes-obj.dir/build.make:1219: CMakeFiles/unity-scopes-obj.dir/src/scopes/internal/OnlineAccountClientImpl.cpp.o] Error 1
```

```
In file included from /tmp/unity-scopes-api/include/unity/scopes/internal/PreviewReplyImpl.h:23,
                 from /tmp/unity-scopes-api/src/scopes/internal/PreviewReplyImpl.cpp:20:
/tmp/unity-scopes-api/include/unity/scopes/internal/ReplyImpl.h:51:18: error: ‘function’ in namespace ‘std’ does not name a template type
     typedef std::function<void()> CleanupFunc;
                  ^~~~~~~~
/tmp/unity-scopes-api/include/unity/scopes/internal/ReplyImpl.h:51:13: note: ‘std::function’ is defined in header ‘<functional>’; did you forget to ‘#include <functional>’?
/tmp/unity-scopes-api/include/unity/scopes/internal/ReplyImpl.h:29:1:
+#include <functional>
 
/tmp/unity-scopes-api/include/unity/scopes/internal/ReplyImpl.h:51:13:
     typedef std::function<void()> CleanupFunc;
             ^~~
make[2]: *** [CMakeFiles/unity-scopes-obj.dir/build.make:1284: CMakeFiles/unity-scopes-obj.dir/src/scopes/internal/PreviewReplyImpl.cpp.o] Error 1
```

```
In file included from /tmp/unity-scopes-api/src/scopes/qt/internal/QScopeBaseAPIImpl.cpp:19:
/tmp/unity-scopes-api/include/unity/scopes/qt/internal/QScopeBaseAPIImpl.h:55:30: error: ‘function’ in namespace ‘std’ does not name a template type
     using FactoryFunc = std::function<QScopeBase*()>;
                              ^~~~~~~~
/tmp/unity-scopes-api/include/unity/scopes/qt/internal/QScopeBaseAPIImpl.h:55:25: note: ‘std::function’ is defined in header ‘<functional>’; did you forget to ‘#include <functional>’?
/tmp/unity-scopes-api/include/unity/scopes/qt/internal/QScopeBaseAPIImpl.h:32:1:
+#include <functional>
 #include <memory>
/tmp/unity-scopes-api/include/unity/scopes/qt/internal/QScopeBaseAPIImpl.h:55:25:
     using FactoryFunc = std::function<QScopeBase*()>;
                         ^~~
/tmp/unity-scopes-api/include/unity/scopes/qt/internal/QScopeBaseAPIImpl.h:60:34: error: expected ‘)’ before ‘const’
     QScopeBaseAPIImpl(FactoryFunc const& creator);
                      ~           ^~~~~~
                                  )
/tmp/unity-scopes-api/include/unity/scopes/qt/internal/QScopeBaseAPIImpl.h:79:5: error: ‘FactoryFunc’ does not name a type
     FactoryFunc qtscope_creator_;
     ^~~~~~~~~~~
/tmp/unity-scopes-api/src/scopes/qt/internal/QScopeBaseAPIImpl.cpp:81:37: error: expected constructor, destructor, or type conversion before ‘(’ token
 QScopeBaseAPIImpl::QScopeBaseAPIImpl(FactoryFunc const& creator)
                                     ^
/tmp/unity-scopes-api/src/scopes/qt/internal/QScopeBaseAPIImpl.cpp: In member function ‘virtual bool unity::scopes::qt::internal::QScopeBaseAPIImpl::event(QEvent*)’:
/tmp/unity-scopes-api/src/scopes/qt/internal/QScopeBaseAPIImpl.cpp:106:33: error: ‘qtscope_creator_’ was not declared in this scope
             qtscope_impl_.reset(qtscope_creator_());
                                 ^~~~~~~~~~~~~~~~
/tmp/unity-scopes-api/src/scopes/qt/internal/QScopeBaseAPIImpl.cpp:106:33: note: suggested alternative: ‘qtscope_impl_’
             qtscope_impl_.reset(qtscope_creator_());
                                 ^~~~~~~~~~~~~~~~
                                 qtscope_impl_
make[2]: *** [src/scopes/qt/CMakeFiles/unity-scopes-qt.dir/build.make:206: src/scopes/qt/CMakeFiles/unity-scopes-qt.dir/internal/QScopeBaseAPIImpl.cpp.o] Error 1
```

</details>

--- a/include/unity/scopes/OnlineAccountClient.h
+++ b/include/unity/scopes/OnlineAccountClient.h
@@ -23,6 +23,7 @@
 #include <unity/scopes/Result.h>
 #include <unity/util/NonCopyable.h>
 
+#include <functional>
 #include <memory>
 
 namespace unity

--- a/include/unity/scopes/internal/AbstractObject.h
+++ b/include/unity/scopes/internal/AbstractObject.h
@@ -21,6 +21,7 @@
 #include <unity/util/DefinesPtrs.h>
 #include <unity/util/NonCopyable.h>
 
+#include <functional>
 #include <mutex>
 
 namespace unity

--- a/include/unity/scopes/internal/Reaper.h
+++ b/include/unity/scopes/internal/Reaper.h
@@ -23,6 +23,7 @@
 
 #include <atomic>
 #include <condition_variable>
+#include <functional>
 #include <list>
 #include <mutex>
 #include <thread>

--- a/include/unity/scopes/internal/ReplyImpl.h
+++ b/include/unity/scopes/internal/ReplyImpl.h
@@ -26,6 +26,7 @@
 #include <unity/scopes/Variant.h>
 
 #include <atomic>
+#include <functional>
 
 namespace unity
 {

--- a/include/unity/scopes/internal/zmq_middleware/ServantBase.h
+++ b/include/unity/scopes/internal/zmq_middleware/ServantBase.h
@@ -23,6 +23,7 @@
 #include <unity/scopes/internal/zmq_middleware/Current.h>
 #include <unity/util/NonCopyable.h>
 
+#include <functional>
 #include <memory>
 #include <string>
 #include <unordered_map>

--- a/include/unity/scopes/qt/internal/QScopeBaseAPIImpl.h
+++ b/include/unity/scopes/qt/internal/QScopeBaseAPIImpl.h
@@ -28,6 +28,7 @@
 
 #include <QtCore/QObject>
 
+#include <functional>
 #include <thread>
 #include <memory>
 #include <atomic>

